src/helpers/serviceSelector.ts

```diff
@@ -0,0 +1,117 @@
+import { ProcessorTransformationRequest, RouterTransformationRequestData } from '../types/index';
+import { INTEGRATION_SERVICE } from '../routes/utils/constants';
+import CDKV1DestinationService from '../services/destination/cdkV1Integration';
+import CDKV2DestinationService from '../services/destination/cdkV2Integration';
+import DestinationService from '../interfaces/DestinationService';
+import NativeIntegrationDestinationService from '../services/destination/nativeIntegration';
+import SourceService from '../interfaces/SourceService';
+import NativeIntegrationSourceService from '../services/source/nativeIntegration';
+import { PlatformError } from '../v0/util/errorTypes';
+import ComparatorService from '../services/comparator';
+
+export default class ServiceSelector {
+  private static serviceMap: Map<string, any> = new Map();
+
+  private static isCdkDestination(destinationDefinitionConfig: Object) {
+    return !!destinationDefinitionConfig?.['cdkEnabled'];
+  }
+
+  private static isCdkV2Destination(destinationDefinitionConfig: Object) {
+    return (
+      process.env.CDK_V2_Enabled === 'true' &&
+      Boolean(destinationDefinitionConfig?.['cdkV2Enabled'])
+    );
+  }
+
+  private static isComparatorEnabled(destinationDefinitionConfig: Object): boolean {
+    return !!destinationDefinitionConfig['camparisonTestEnabeld'];
+  }
+
+  private static getSecondaryServiceName(destinationDefinitionConfig: Object): string {
+    return destinationDefinitionConfig['camparisonService'];
+  }
+
+  private static fetchCachedService(serviceType: string) {
+    if (this.serviceMap.has(serviceType)) {
+      return this.serviceMap.get(serviceType);
+    }
+    switch (serviceType) {
```

Comment:
I am seeing same set of values which match the cases. Rather than this, we can achieve this in a very simple manner

```javascript
const validServicesMap = {
 INTEGRATION_SERVICE.CDK_V1_DEST: CDKV1DestinationService,
 INTEGRATION_SERVICE.CDK_V2_DEST: CDKV2DestinationService,
 INTEGRATION_SERVICE.NATIVE_DEST: NativeIntegrationDestinationService,
 INTEGRATION_SERVICE.NATIVE_SOURCE: NativeIntegrationSourceService
}

private static fetchCachedService(serviceType: string) {
    if (!this.serviceMap.has(serviceType)) {
      // One more variation -> You can evaluate serviceTypeMap.keys() only once as it is not going to change in run-time
      if (serviceTypeMap.has(serviceType)) {
        const Service = serviceTypeMap[serviceType];
        this.serviceMap.set(
          INTEGRATION_SERVICE.NATIVE_DEST,
          new Service(),
        );
      }
    }
    return this.serviceMap.get(serviceType);
}
```

Reply:
A similar pattern can be implemented wherever such kind of switch cases or if-else combinations are seen. The goal to make sure that the logic doesn't look too bulky and is very readable. Please check if this pattern can be applied anywhere else 🙂

Reply:
This is addressed